Assets/Scripts/Structres/Projectile.cs

```diff
@@ -0,0 +1,46 @@
+using UnityEngine;
+using FishNet.Object;
+
+public class Projectile : NetworkBehaviour
+{
+    [SerializeField] private float _speed = 1f;
+    [SerializeField] private float _damage;
+
+    private Transform _target;
+
+    public Transform Target { set => _target = value; }
+
+    private void Awake() => Destroy(gameObject, 10f);
+
+    private void FixedUpdate()
+    {
+        if (!IsServer)
+            return;
+
+        Move();
+
+        if (Vector3.Distance(transform.position, _target.position) < 0.1f)
```

Comment:
Have you measured whether or not this is more optimised than a physics linecast? Or if we could measure the distance without using the squared root.